Cubical/Foundations/Logic.agda

```diff
@@ -273,3 +277,96 @@ Decₚ P = Dec [ P ] , isPropDec (snd P)
 ⊓-∀-distrib P Q =
   ⇒∶ (λ {(p , q) a → p a , q a})
   ⇐∶ λ pq → (fst ∘ pq ) , (snd ∘ pq)
+
+--------------------------------------------------------------------------------
+-- Introduce the "powerset" of a type in the style of Escardó's lecture notes:
+-- https://www.cs.bham.ac.uk/~mhe/HoTT-UF-in-Agda-Lecture-Notes/HoTT-UF-Agda.html#propositionalextensionality
+
+ℙ : ∀ {ℓ} → Type ℓ → Type (ℓ-suc ℓ)
+ℙ {ℓ} X = X → hProp ℓ
+
+_∈_ : {X : Type ℓ} → X → ℙ X → Type ℓ
+x ∈ A = [ A x ]
+
+_⊆_ : {X : Type ℓ} → ℙ X → ℙ X → Type ℓ
+A ⊆ B = ∀ x → x ∈ A → x ∈ B
+
+∈-isProp : {X : Type ℓ} (A : ℙ X) (x : X) → isProp (x ∈ A)
+∈-isProp A x = (A x) .snd
+
+⊆-isProp : {X : Type ℓ} (A B : ℙ X) → isProp (A ⊆ B)
+⊆-isProp A B = isPropΠ2 (λ x _ → ∈-isProp B x)
+
+⊆-refl : {X : Type ℓ} (A : ℙ X) → A ⊆ A
+⊆-refl A x = idfun (x ∈ A)
+
+
+⊆-refl-consequence : {X : Type ℓ} (A B : ℙ X)
+                   → A ≡ B → (A ⊆ B) × (B ⊆ A)
+⊆-refl-consequence {X} A B p = subst (λ B → (A ⊆ B)) p (⊆-refl A) , subst (λ A → (B ⊆ A)) (sym p) (⊆-refl B)
+
+
+⊆-extensionality : {X : Type ℓ} (A B : ℙ X)
+                 → (A ⊆ B) × (B ⊆ A) → A ≡ B
+⊆-extensionality A B (φ , ψ) i x = ⇔toPath {P = (A x)} {Q = (B x)} (φ x) (ψ x) i
+
+
+powersets-are-sets : {X : Type ℓ} → isSet (ℙ X)
+powersets-are-sets {X = X} = isSetΠ (λ _ → isSetHProp)
```

Comment:
Maybe it would be nice to add `isSet→` and friends to avoid having to write `λ _ →` all the time

Reply:
I added an issue about this